### PR TITLE
update relay python api doc

### PR DESCRIPTION
**Pull request by yongwww**
Fix for issue #2644

### CODE REVIEW

**Comment by yongwww** [comment]
@tqchen @zhiics @wweic @yzhliu pls help take a look. Not sure if the code snippet below works (if not, pls suggest... sorry I am not familiar with Sphinx), trying to link to relay op list doc, it makes no sense to maintain two copies

 ```
   .. toctree::
   :maxdepth: 2

   langref/relay_op
```

File: docs/api/python/relay/nn.rst [hunk 1]
```diff
@@ -1,7 +0,0 @@
-tvm.relay.nn
```
Comment (tqchen) [inline docs/api/python/relay/nn.rst]: We do need t make keep relay/nn.rst and relay/vision.rst

**Comment by tqchen** [comment]
For now, let us not cross ref, let us keep the relay.nn and relay.vision namespace because they are what are used directly by the user. Let us not refer to relay_op for now, since that will result in a too deep nesting.

You can build the doc locally to verify

We can bring other changes.

**Comment by wweic** [comment]
@yongwww after you make the change, try following command to build docs locally(from Jenkins script). Then verify the docs in browser.

````shell
mkdir -p build
cd build
cp ../cmake/config.cmake .
echo 'set(USE_SORT' 'ON)' >> config.cmake
echo 'set(USE_GRAPH_RUNTIME_DEBUG' 'ON)' >> config.cmake
echo 'set(USE_LLVM' 'llvm-config-4.0)' >> config.cmake
echo 'set(USE_NNPACK' 'ON)' >> config.cmake
echo 'set(NNPACK_PATH' '/NNPACK/build/)' >> config.cmake
echo 'set(USE_ANTLR' 'ON)' >> config.cmake
echo 'set(CMAKE_CXX_COMPILER' 'g++)' >> config.cmake
echo 'set(CMAKE_CXX_FLAGS' '-Werror)' >> config.cmake
cd ..
docker/bash.sh tvmai/ci-cpu ./tests/scripts/task_build.sh build -j2
docker/bash.sh tvmai/ci-cpu ./tests/scripts/task_python_docs.sh
````

**Pull request merged**